### ISSUE TITLE
Feature request: GitHub IdP Teams Sync support

### ISSUE THREAD

**Issue by sean-nixon** [terraform-providers/terraform-provider-github#293]
## Overview

We are wanting to leverage GitHub's new Team Sync capabilities to manage GitHub team memberships with groups from our IdP (Azure AD). It would be great if the GitHub provider supported that functionality. 

## Proposed new resources
This functionality would probably require two new resources/data sources at a minimum.
* New resource: `github_team_sync_group_mapping`
* New data source: `github_team_sync_groups`

## Possible configuration:
```hcl
data "github_team_sync_groups" "my_groups" {}

resource "github_team" "some_team" {
  name        = "SomeTeam"
  description = "Some cool team"
}

resource "github_team_sync_group_mapping" "team_mapping" {
  team_id  = github_team.some_team.id
  
  dynamic "group" {
    for_each = [for g in data.my_groups.groups : g if g.name == "some_team_group"]
    content {
      group_id          = each.value.group_id
      group_name        = each.value.group_name
      group_description = each.value.group_description
    }
  } 
}
```

## Related materials
More information on Team Sync here
https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/synchronizing-teams-between-your-identity-provider-and-github#enabling-team-synchronization

API Documentation
https://developer.github.com/v3/teams/team_sync/#create-or-update-idp-group-connections

**Comment by jopedros** [comment]
Hi @anGie44 

Are you going to work on this one? If not, I could help.

**Comment by jlpedrosa** [comment]
Hi @anGie44 

Are you working on this one? If help is needed I could take it.

**Comment by anGie44** [comment]
hi @jlpedrosa! I am atm. I'll reach out if anything/reference you to the forthcoming PR if interested :)

**Comment by anGie44** [comment]
Hi @sean-nixon @jlpedrosa and all following this issue, I've proposed changes with the linked PR but could use help in running acceptance tests. If you have access to an enterprise cloud test org and able to configure SAML and complete the Azure AD admin OAuth flow that would be of much help to validate the suggested changes :)

**Comment by davidkarlsen** [comment]
Any update on this?

**Comment by anGie44** [comment]
hi @davidkarlsen, we plan to have this feature in the upcoming `v.2.8.0` release of the provider

**Comment by anGie44** [comment]
Note: related changes have been merged and will release with the `v2.8.0` release of the provider
